modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/components/CompletionRateCircle.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.graphics.drawscope.withTransform
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH10
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH50
+import au.com.shiftyjelly.pocketcasts.endofyear.utils.rainbowBrush
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import com.airbnb.android.showkase.annotation.ShowkaseComposable
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+private val PercentFontSize = 80.sp
+private const val CompletionCircleAvailableWidthPercent = .75f
+private val CompletionCircleBaseColor = Color(0xFF292B2E)
+
+@Composable
+fun CompletionRateCircle(
+    percent: Int,
+    titleColor: Color,
+    subTitleColor: Color,
+    modifier: Modifier = Modifier,
+) {
+    BoxWithConstraints(
+        modifier = modifier,
+        contentAlignment = Alignment.Center,
+    ) {
+        val availableWidth = maxWidth * CompletionCircleAvailableWidthPercent
+        Box(
+            modifier = Modifier
+                .size(availableWidth)
+                .drawCompletionCircle(percent),
+            contentAlignment = Alignment.Center,
+        ) {
+            CompletionTextContent(
+                percent = percent,
+                titleColor = titleColor,
+                subTitleColor = subTitleColor,
+            )
+        }
+    }
+}
+
+@Composable
+private fun CompletionTextContent(
+    percent: Int,
+    titleColor: Color,
+    subTitleColor: Color,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        TextH10(
+            text = "$percent%",
+            color = titleColor,
+            fontFamily = StoryFontFamily,
+            fontSize = PercentFontSize,
+            fontWeight = FontWeight.W300,
+            maxLines = 1,
+            textAlign = TextAlign.Center,
+        )
+        TextH50(
+            text = stringResource(LR.string.end_of_year_stories_year_completion_rate),
+            color = subTitleColor,
+            fontFamily = StoryFontFamily,
+            fontSize = 14.sp,
+            fontWeight = FontWeight.W600,
+            textAlign = TextAlign.Center,
+            modifier = Modifier.offset(y = (-5).dp)
```

Comment:
I might want to try out `BoxWithConstraints` to align and size the texts. I added a subtask to update fontSize in another story: https://github.com/Automattic/pocket-casts-android/pull/1495#discussion_r1376655773. I'll check this one out too.